docs/extensions/guides/stores.md

```diff
@@ -50,7 +50,7 @@ export class ExamplePreferencesStore extends Store.ExtensionStore<ExamplePrefere
   }
 }
 
-export const examplePreferencesStore = ExamplePreferencesStore.getInstance<ExamplePreferencesStore>();
+export const examplePreferencesStore = ExamplePreferencesStore.getInstanceOrCreate();
```

Comment:
Do we want to introduce this kind of breaking change in a minor version? 🤔

Reply:
Hmm... or is this PR actually backwards compatible?

Reply:
I guess it is a breaking change because I seperated out when an instance of a class would be created. I did this so that the program doesn't get subtly broken if someone calls `resetInstance`. I guess in the general case we could call the constructor if it has not arguments declared.

Note: I removed the `export const clusterStore = ...`, `export const userStore = ...`, and others because those are inherantly not singletons at all. That is technically a breaking change. Several places I have had to fix some implicit assumptions about the order in which items are created because of this. But I think that this leaves the codebase in a better and more robust state.

Lastly, if we wanted to we could hold onto this until after 4.1 (since 4.1 includes the auto update confirmation dialog PR). And then immediatly release 5.0 since this would be allowed for a major version.

Reply:
> Lastly, if we wanted to we could hold onto this until after 4.1 (since 4.1 includes the auto update confirmation dialog PR). And then immediatly release 5.0 since this would be allowed for a major version.

We are not planning to release 5.0 that fast. Any possibility to make this without breaking the extension API compatibility?

Reply:
I think that undo the breaking change is to re-add the `export const clusterStore` et al. but mark them as deprecated. The other parts of this PR are not actual breaking the API, just when things get instantiated.

Reply:
Actually, now that I think about it. I don't think we can really go about adding back the `export const clusterStore` et al. Because those would have to be `export const clusterStore = ClusterStore.getInstanceOrCreate();` which somewhat nullifies these changes (since imo it is preferable to have as few places as possible for the creation of these singletons)

Reply:
PR now targets 5.0 so we can break the extension API compatibility.